### ISSUE TITLE
In Notepad++ v7.7 scroll synchronization does not work

### ISSUE THREAD

**Issue by andrecool-68** [pnedev/compare-plugin#201]
In the new version of Notepad++ v7.7 scroll synchronization does not work during file comparison (plugin ComparePlugin2.0)

Notepad++ v7.7   (64-bit)
Build time : May 14 2019 - 01:26:08
Path : C:\Users\Uzeer\Desktop\npp.7.7.bin\x64\notepad++.exe
Admin mode : ON
Local Conf mode : ON
OS : Windows 7 (64-bit)
Plugins : ComparePlugin.dll DSpellCheck.dll HTMLTag.dll JSMinNPP.dll mimeTools.dll NppConverter.dll NppExport.dll NppMarkdownPanel.dll VisualStudioLineCopy.dll XMLTools.dll _CustomizeToolbar.dll

**Comment by pnedev** [comment]
@andrecool-68 ,

Strange,
I couldn't reproduce the issue on Notepad 7.7 (32-bit under Wine) neither with Compare v2 nor with ComparePlus.
I'll try again on native Windows and with 64-bit version.
Thanks for the feedback :+1:

**Comment by andrecool-68** [comment]
@pnedev 
![1](https://user-images.githubusercontent.com/3471892/57853868-855f1180-77ef-11e9-8f49-af7189796446.gif)

**Comment by pnedev** [comment]
@andrecool-68 , @rddim ,

I didn't test Compare v2 I consider it kind of "obsolete" now that I'm close to releasing ComparePlus.

But I did test with ComparePlus and the issue is present only on 64-bit Notepad++ 7.7.
That's kind of normal considering the variable types and notifications changes in the new Scintilla.

I'll fix it some time soon, thanks again for letting me know about the problem :+1:

**Comment by fkl79675** [comment]
I experienced the same issue. Thank you pnedev for working on a fix and an all new version.

(@andrecool-68 How did you create this gif?)

**Comment by andrecool-68** [comment]
@fkl79675 https://github.com/NickeManarin/ScreenToGif

**Comment by Stady234** [comment]
I closed my duplicate issue.. sorry.  Interested in the new version!

**Comment by pnedev** [comment]
Thanks @Yaron10 :+1:

**Comment by Pyre909** [comment]
Hi @pnedev,

I tested this in the commit **Update to Notepad++ 7.7 API**. It's working for me, I don't know what others are experiencing. I have only tested this only comparing with XML files.

Debug Info:
Notepad++ v7.7   (64-bit)
Build time : May 25 2019 - 19:46:35
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 10 (64-bit)
Plugins : ComparePlugin.dll LuaScript.dll mimeTools.dll NppConverter.dll

**Comment by pnedev** [comment]
Thanks for testing and sharing @Pyre909 .
It should be working for everybody.

The issue is fixed for ComparePlugin v2, here are the [**x86**](https://ci.appveyor.com/api/buildjobs/go5h6x448jg551v8/artifacts/ComparePlugin.dll) and **[x64](https://ci.appveyor.com/api/buildjobs/g4bidp2g30xf795h/artifacts/ComparePlugin.dll)** binaries.

**Comment by pidgeon777** [comment]
Hello pnedev, just a question, does this new release include all of the fixes you mentioned during these last days? For example, undo history bug, live compare feature, etc.

Thanks again.

**Comment by pnedev** [comment]
Hello @pidgeon777 ,

No, it is the official Compare 2.0 but with changed Notepad++ API so the people who moved to Notepad++ 7.7 can have a working Compare.

The fixes you are talking about will be in the next official release which is going to be completely new plugin (it will be called ComparePlus). You can read more about that [**here**](https://github.com/pnedev/compare-plugin/issues/195).

BR

**Comment by pidgeon777** [comment]
Cool, thank you!

**Comment by dhanotra** [comment]
Enable this while doing compare.
View -> Synchronize Vertical Scrolling

**Comment by b-morgan** [comment]
I'm confused. 

> Thanks for testing and sharing @Pyre909 .
> It should be working for everybody.
> 
> The issue is fixed for ComparePlugin v2, here are the [**x86**](https://ci.appveyor.com/api/buildjobs/go5h6x448jg551v8/artifacts/ComparePlugin.dll) and **[x64](https://ci.appveyor.com/api/buildjobs/g4bidp2g30xf795h/artifacts/ComparePlugin.dll)** binaries.

I downloaded the x64 binary and installed it in the Plugins folder, opened Notepad++ 7.7.1 (64-bit), and started a compare. The Synchronize Vertical Scrolling keeps getting turned off. Did I misunderstand "the issue is fixed"?

**Comment by pnedev** [comment]
Don't do that.

If your Notepad++ version is 7.7 or higher use these Compare 2.0 versions:

x86: https://github.com/pnedev/compare-plugin/releases/download/v2.0.0_npp7.7/ComparePlugin_v2.0.0_npp7.7_x86.zip

x64: https://github.com/pnedev/compare-plugin/releases/download/v2.0.0_npp7.7/ComparePlugin_v2.0.0_npp7.7_X64.zip

**Comment by broszkiet** [comment]
Uninstall  Compare plugin using Plugins Admin and install again.

**Comment by ballios** [comment]
I confirm that in Notepad++ v7.8.1 scroll synchronization does not work (for ComparePlugin v2.0.0 npp7.7 x64)!
Please fix it to make it convenient to work!

**Comment by pnedev** [comment]
Does anyone else experience the same issue as @ballios ?

**Comment by Stady234** [comment]
Mine works... 
![image](https://user-images.githubusercontent.com/32916064/68169266-160de300-ff31-11e9-9af7-713d9bf11b0b.png)

**Comment by pnedev** [comment]
Thanks @Stady234 .

@ballios ,
Can you please double check it doesn't work on your side?

**Comment by ballios** [comment]
> Can you please double check it doesn't work on your side?
I confirm that in Notepad++ v7.8.1 **scroll synchronization** does not work.
I need to switch on "Sync Verticle Scrolling" button manually...

**Comment by ballios** [comment]
> Can you please double check it doesn't work on your side?
I confirm that in Notepad++ v7.8.1 **scroll synchronization** does not work.
I need to switch on "Sync Verticle Scrolling" button manually...

**Comment by ballios** [comment]
> Thanks @Stady234 .
> 
> @ballios ,
> Can you please double check it doesn't work on your side?
I confirm that in Notepad++ v7.8.1 **scroll synchronization** does not work.
I need to switch on "Sync Verticle Scrolling" button manually...

**Comment by pnedev** [comment]
Ok, I'll check that

**Comment by ballios** [comment]
Moreover, your **ComparePlugin v2.0.0 npp7.7 x64** (there is no such problem in default ComparePlugin v2.0.0) does not always work well when comparing long lines, for example code. The impossibility of scrolling with the mouse appears, and comparing the contents is somehow stuck.

**Comment by pnedev** [comment]
@ballios ,

I just checked with the portable Notepad++ 7.8.1 version.
I used the Notepad++ Plugin's Admin to download Compare v2 and it is working fine.
I didn't experienced any problems with the scroll sync.

Besides, the only difference between Compare v2 and Compare v2 for Npp 7.7 and above is the changes in the plugins API that comes from the Notepad++ 7.7 new plugins handling.

If you need further help you'll need to describe in details what you are doing and how - step by step starting from Notepad++ 7.8.1 installation then Compare v2 installation and so on.

**Comment by dtwaling** [comment]
Just want to confirm (in case anyone else checks this thread) --> when updating npp 7.x to version 7.8.1 or later, you can use Plugin Admin to remove the plugin, then open Plugin Admin to select and re-installing the Compare plugin again.  This does resolve the issue described.

This is not totally unexpected given the API changes in npp.  And, I did have to do this for a few other plugins as well after updating to npp v7.8.x

**Comment by JohanPirlouit** [comment]
Hello everyone,

As @dtwaling said, I can confirm that to recover the vertical scroll sync with NPP 7.8+, I had to reinstall Compare plugin using the Plugin Admin (remove and reinstall) and ... it works!

Thanx @dtwaling for the tip ;-)

**Comment by pnedev** [comment]
And one more thing:

If you continue to experience problems with vertical scroll sync while in compare and you have followed all the instructions above (to update the Compare v2 plugin) then **remove all your other plugins** because they might not be updated to the new N++ 7.7 (and above) API and thus compromise Notepad++ normal work.

Leave **only** the updated Compare v2 for Npp7.7 plugin and retry.

**Comment by Corendiel** [comment]
I'm in version 7.8.4 of Notepad++ and 2.0.0 of compare plugin.
I remove every other plugin and Compare is still not activating the vertical sync scrolling.
![image](https://user-images.githubusercontent.com/1401010/74351296-f171a480-4d84-11ea-82fc-184fe6ae699b.png)
Even if I start in 2 views mode with vertical sync scrolling activated.
Turning on Compare will turn off vertical sync scrolling and reactivate it after I stop Compare.

So you might be right Compare use its own vertical sync scrolling setting. The problem is that the option is active by default and it doesn't save the setting for future use.

**Comment by Stady234** [comment]
Has to be something else you have... I just installed and checked on 5 of my work machines that never had notepad++ on them... everything works as expected.  

I would remove notepad++ and all plugins, then follow up with a "system maintenance" tool... i have had great luck with Glarysoft.com (and it is free)...

**Comment by pnedev** [comment]
@Corendiel ,

I need a little clarification:

When you say
> I remove every other plugin and Compare is still not activating the vertical sync scrolling.

do you mean that the files are not scrolling vertically together while they are compared?
Or you mean that the Notepad++ vertical scroll sync option is off?

> So you might be right Compare use its own vertical sync scrolling setting.

Yes, I'm sure. I'm the developer of Compare plugin :)

> The problem is that the option is not active by default and it doesn't save the setting for future use.

Again, what option is not active by default and saved for future use?
Do you mean the Notepad++ vertical scroll sync option?

-------------------------------------

To summarize again:
Compare plugin **does not** rely at all on Notepad++ vertical scroll sync setting.
Compare plugin **does** the vertical sync on **its own** and it does it **always**. It cannot be turned off.
In order not to interfere with Notepad++, Compare plugin **temporarily switches off** Notepad++ vertical sync **while in compare**. After the compare is closed the Notepad++ vertical sync is **restored** to its state before the compare.

BR

**Comment by Corendiel** [comment]
Thanks @pnedev, for looking into this. I like your plugin and that's why I'm taking the time to document the issue here.

For info, I have this issue on 2 different computers one at home and my work laptop. Both are on Windows 10. At work, it is Windows 10 Enterprise Version 1809 Build 17763.914.
I can give you my home version later if needed. I'm writing from work right now.
Notepad++ 64 version 7.8.4 and Compare Plugin 2.0.0
It might be a weird combination of upgrades that caused it but seeing the number of people reporting a problem might not be a rare occurrence and point out that this feature has some weaknesses.

I removed every other plugin from the plugin folder before doing the following test.

I open Notepad++ with 2 long enough text files in side by side mode.
The Sync Vertical Scrolling is not active and not showing as such in the Toolbar.
![image](https://user-images.githubusercontent.com/1401010/74457214-f2bed200-4e55-11ea-86bb-d173d7a5886a.png)

I enable it in the Toolbar and it starts to Sync Scroll Correctly
![image](https://user-images.githubusercontent.com/1401010/74457469-48937a00-4e56-11ea-897e-f763bd669e1e.png)

I close Notepad++

I restart Notepad++ and it opens my last session with my 2 files side by side. The Sync Vertical Scroll is disabled again. Notepad++ didn't save the status of my option before closing.
![image](https://user-images.githubusercontent.com/1401010/74457602-7c6e9f80-4e56-11ea-8e1f-b9e5da2b4ebe.png)

I start a Compare. The Sync Horizontal Scroll turns on in the Toolbar but not the **Vertical** Scroll and Sync Vertical Scrolling doesn't work.
![image](https://user-images.githubusercontent.com/1401010/74457926-ee46e900-4e56-11ea-85d8-a8429885d701.png)
I click the Sync Vertical Scroll in the Toolbar and it starts scrolling correctly. I can turn it on and off.
![image](https://user-images.githubusercontent.com/1401010/74458430-af656300-4e57-11ea-9982-7b4f8b6b509a.png)
I stop the Compare by using the "Clear all Compares" option in the Plugin menu. Both Vertical and Horizontal Sync Scroll turn off.
![image](https://user-images.githubusercontent.com/1401010/74458737-226ed980-4e58-11ea-9f4a-fafab1d7111a.png)

I turn on the Sync Vertical Sync in the Toolbar. It works again.
![image](https://user-images.githubusercontent.com/1401010/74458836-4b8f6a00-4e58-11ea-9db4-71bbc8d7f88e.png)

I start another Compare. The Vertical scroll turns off, the Horizontal turns on. Vertical Sync doesn't work.
![image](https://user-images.githubusercontent.com/1401010/74458883-5ba74980-4e58-11ea-873f-d50102b504aa.png)

I stop the Compare by using the "Clear all Compares" option in the Plugin menu. Vertical Sync turns back on and Horizontal Sync Scroll turns off. Notepad++ remembers the status of the sync options before the Compare operation.
![image](https://user-images.githubusercontent.com/1401010/74459191-e5efad80-4e58-11ea-9e34-a6d39ea4d2a7.png)

I reinstall Notepad++ after removing the Compare plugin manually. Reinstalled Compare using the Plugin Manager.
It did change somewhat for the better but not completely as expected.
Notepad++ is still not saving the Vertical Sync Scroll option when closing.
When starting a Compare when the option is turned on make the Toolbar button change status.
![image](https://user-images.githubusercontent.com/1401010/74461033-df166a00-4e5b-11ea-9085-269cd19560e6.png)
![image](https://user-images.githubusercontent.com/1401010/74461051-e50c4b00-4e5b-11ea-92bc-edda0ced4a1b.png)
Only the Vertical though.
The Vertical Sync is now working by default but I can no longer turn it on and off. It is just always active no matter the Tool Bar Button status. I don't know if it's the expected behavior or not.
I can turn Horizontal sync on and off but not Vertical sync.

I can live with this new behavior hopping it is not gonna change again. The sync is now always active in Compare mode and that is the main reason for me to use it in the first place.

Let me know if you have any more questions or if you need anything else.

**Comment by Stady234** [comment]
Looks like you are using the wrong buttons completely... as in you have no idea how to use the plugin.

open two files you want to compare... don't change any options  AT ALL.. don't click anything BUT the compare button on the toolbar....  Not the vertical sync, not the horizontal sync... NOTHING....

Then show what happens....  I use the compare tool almost daily when looking for differences in 3dprinter config files... i can make a few changes, and reclick the same button, and it will resync the compare process and update the list...  No other issues....

**Comment by pnedev** [comment]
Thanks for trying to help @Stady234 .

Hi @Corendiel ,

Thank you for the detailed explanation.
I really don't know what else to add to the summary in my previous post.

Notepad++ and Compare use different vertical scroll approaches internally.
That's why Compare turns off Notepad++ vertical scroll while compare is active and restores the previous Notepad++ state (the one before the compare was started) when compare is over.
While in compare you cannot turn off vertical scroll - it is always active. You shouldn't try using Notepad++ vertical scroll while in compare.
Compare plugin v2 has always behaved that way - it never changed.

The reason why you didn't have vertical scroll before you updated Compare plugin is because there are two Compare v2 variants - one is for Notepad++ versions < 7.7 and one is for Notepad++ versions >= 7.7. You can have a look here:
https://github.com/pnedev/compare-plugin/releases/tag/v2.0.0 (for Notepad++ < 7.7)
https://github.com/pnedev/compare-plugin/releases/tag/v2.0.0_npp7.7 (for Notepad++ >= 7.7).

So what happened was that you upgraded Notepad++ from version < 7.7 to version 7.8.4 (> 7.7) but the Compare plugin was not updated and it doesn't work normally because it is not for the new Notepad++ version.
That's why after you reinstalled Compare the proper plugin version was installed (for your new Notepad++) and everything works as expected (you always have vertical scroll while in compare, no matter the Notepad++ vertical scroll setting).

---------------------------------------------

As a side note:
Notepad++ vertical scroll not being preserved when Notepad++ restarts is Notepad++'s issue and has nothing to do with Compare plugin.

**Comment by Corendiel** [comment]
@Stady234 I have been using this plugin for 10 years I know how to use it thank you. It's not because you use it a certain way that it's not intended to work in another expected way. Buttons are meant to be clicked. Again I'm, not the only person reporting this issue. There are other threads out there for this 
https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5250
https://sourceforge.net/p/npp-compare/bugs/38/

@pnedev, Thanks for the versioning explanation. I believe this is the true answer. Looking back at the history you gave that explanation on Jul 26, 2019.
It is confusing though that the Compare plugin didn't change in version number. I don't know if it's complex for you to increase the version like 2.0.1. It might force more people to upgrade to the right version. I ended in this exact messed up config on 2 very different PCs not updated at the same time. Looking at some comment after Jul 26 other people didn't see the _npp7.7 in the version.
I believe the issue is fixed Uninstalling the Compare plugin using Plugins Admin and installing it again should do the trick. People should not trust the version number and reinstall anyway.

I never really tinkered with the vertical scrolling before until it started not beeing active by default. So I believe you if you say it has always been like that. However, now that I tested thoroughly I still believe the comportment of the toolbar buttons is confusing.

- First, Horizontal and Vertical do not behave the same way. Horizontal turn On while Vertical turns Off.

- Second, even if Compare uses its own sync it does change the status of the Notpad++ Sync buttons. Making any average user think they are linked somehow. Maybe they should be greyed out in compare mode.

- Finally, why is the sync always On in Compare? and why not use Notepad++ Sync? I still have the wrong setup at home that I have still not fixed, the Notepad++ sync seems to take over and I'm able to turn it on and off and it behaves as expected while in Compare mode. The only issue is the sync is disabled by default and not saving my preference from one time to the next.
![image](https://user-images.githubusercontent.com/1401010/74541007-cd41cf00-4f0e-11ea-91a6-4308a0653774.png)

Looks like you are working on a ComparePlus plugin so you might have a different approach. If you need an annoying perfectionist tester let me know.

**Comment by pnedev** [comment]
@Corendiel ,

Thanks for the suggestions and I'm glad we found the root cause. :+1: 

One remark:
https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5250 is a thread for Notepad++ vertical scroll setting not being remembered correctly not for the Compare plugin. So it is a completely different issue.

And now some clarifications:

>  I don't know if it's complex for you to increase the version like 2.0.1.

It is not complex at all of course.
But it is the same Compare plugin version (v2), nothing changed in its functionality.
It is the Notepad++ that actually changed its plugins API since v7.7 making **almost all current plugins obsolete**. The change was for good of course but Notepad++ and the plugins out there are developed by different people and the synchronization when such things happen is kind of lacking.
The normal thing in this case would be when Notepad++ triggers its automatic update to notify the user that his/hers plugins are no longer safe to use.
The normal thing to report as an issue in this case would be to say "After updating Notepad++ to version X, Compare vertical scroll is no longer working". It was working before, right, so what changed?
I didn't want to change the version number because it would be confusing as well.
If we look at it the other way around, you are still at Notepad++ version 7.5 and you see that there is a new Compare plugin version 2.0.1. You then might want the latest version and if you update only the Compare plugin you'll get to a similar situation because 2.0.1 would not be compatible with N++ 7.5.
I don't know which of the two is the better approach but here we are.

> First, Horizontal and Vertical do not behave the same way. Horizontal turn On while Vertical turns Off.

They are not meant to behave the same way.
Besides, Compare relies on Horizontal scrolling for user convenience - it doesn't need special handling for horizontal scrolling.

> Second, even if Compare uses its own sync it does change the status of the Notpad++ Sync buttons. Making any average user think they are linked somehow. Maybe they should be greyed out in compare mode.

There is no other way for Compare to switch off Notepad++ vertical sync - the buttons state will change inevitably. I agree this might be confusing. ComparePlus tries to disable the Notepad++ scroll buttons to block user interaction.

> Finally, why is the sync always On in Compare? and why not use Notepad++ Sync?

I will answer you by quoting you:

>  It's not because you use it a certain way that it's not intended to work in another expected way.

If the left and right document panes are not of equal horizontal size and you have WordWrap turned on, Notepad++ vertical scroll sync will not work properly to align compared lines.
The user might as well disable vertical scroll, scroll one of the documents and then re-enable the vertical scroll. Then the scroll sync will be at different lines of the documents that will break compare diffs alignment.

> Looks like you are working on a ComparePlus plugin so you might have a different approach. If you need an annoying perfectionist tester let me know.

I'll have that in mind, thank you :+1:

**Comment by Corendiel** [comment]
> One remark:
> notepad-plus-plus/notepad-plus-plus#5250 is a thread for Notepad++ vertical scroll setting not being remembered correctly not for the Compare plugin. So it is a completely different issue.
I know it's a different issue but other people on that thread were bringing Compare behavior change too. Like me, most people were not using these buttons and sync scroll features until it stopped being active by default in Compare. My point was I'm not an isolated case and other people are in this mixed match version situation.

> If we look at it the other way around, you are still at Notepad++ version 7.5 and you see that there is a new Compare plugin version 2.0.1. You then might want the latest version and if you update only the Compare plugin you'll get to a similar situation because 2.0.1 would not be compatible with N++ 7.5.
I would expect Plugins Admin to help you pick the right version of the plugin for the right version of your Notepad++. I have no idea how all the versioning works and which team does what but it's probably something the Plugins Admin team should look into.

>If the left and right document panes are not of equal horizontal size and you have WordWrap turned on, Notepad++ vertical scroll sync will not work properly to align compared lines.
The user might as well disable vertical scroll, scroll one of the documents and then re-enable the vertical scroll. Then the scroll sync will be at different lines of the documents that will break compare diffs alignment.
Ok I understand now. I tested too on my messed up setup and it can quickly get crazy. So I guess the best approach would be to have these buttons greyed out or always ON no matter if you click on them so the user knows it's meant to be always on when using compare. Making sure the user gets the proper feedback to reduce confusion.

>I don't know which of the two is the better approach but here we are.
Personally I think people in the most recent version should have the best experience and the least amount of Bugs. Else, in time you discourage people to upgrade. If the guy is in 7.5 and 2.0.1 and has issues, it's unfortunate but he should be as much thorough in updating his Notepad++ as he is with his plugins. 
If you have issues, it's somewhat expected of you to update. In the current situation, it is not intuitive for people in the most recent version to remove and reinstall when they think they are as up to date as possible.

**Comment by pnedev** [comment]
> My point was I'm not an isolated case and other people are in this mixed match version situation.

Yes, of course. All the people upgrading their Notepad++ from version < 7.7 to later versions have had the same problem with Compare and needed to reinstall it. It is confusing no doubt.
That's why this issue was opened and visited by so many people.

> I would expect Plugins Admin to help you pick the right version of the plugin for the right version of your Notepad++. I have no idea how all the versioning works and which team does what but it's probably something the Plugins Admin team should look into.

Plugin Admin picks the right version when you install a plugin anew. It does nothing with already installed plugins. I'm not sure there is a Plugin Admin team. Some people voluntarily help to maintain the plugin list updated with the latest versions of the available plugins for each Notepad++ release.
I don't know more.

> So I guess the best approach would be to have these buttons greyed out or always ON no matter if you click on them so the user knows it's meant to be always on when using compare.

Unfortunately they cannot be ON while in compare because if Notepad++ vertical scroll sync button is ON the sync will be active and it will mess things up. I don't work anymore on Compare plugin, I'm close to releasing the new ComparePlus and it disables those buttons while in compare.

> In the current situation, it is not intuitive for people in the most recent version to remove and reinstall when they think they are as up to date as possible.

I agree but as I said earlier, currently there is no such synchronization between Notepad++ and the plugins versioning. A lot of the plugins are actually not maintained anymore.
If I hadn't updated Compare it would have simply stopped working after Notepad++ 7.7 but you would continue having it installed (as is the case now).
I got your point, I will consider releasing Compare 2 for N++ >= 7.7 with newer version number.

**Comment by penguinairlines** [comment]
I had this problem, found this github issue first google result, uninstall and reinstall the plugin making sure to launch NPP as admin, and vertical scroll works. I did not have to enable any other vertical scroll behavior for NPP. In fact, my only non-default setting of NPP is the cloud directory for portable installation and backup purposes.

**Comment by Wayfarer-es** [comment]
I just confirm what @broszkiet, @Corendiel, and @penguinairlines have already said: I had the same issue and it has been **solved by just uninstalling the Compare plugin** using Plugins Admin and then **installing it again**.

Using Notepad++ 7.9 64-bit with Compare Plugin v2.0.1 on Windows 10 Pro 64-bit 1919 Build 18363.1082
